tool/tsh/common/git_config.go

```diff
@@ -0,0 +1,178 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package common
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/gravitational/trace"
+)
+
+// gitConfigCommand implements `tsh git config`.
+//
+// This command internally executes `git` commands like `git config xxx`. You
+// can generally assume the user has `git` binary installed (otherwise there is
+// no point using the `git` proxy feature). An alternative is to use the
+// `go-git` library.
+type gitConfigCommand struct {
+	*kingpin.CmdClause
+
+	action string
+}
+
+const (
+	gitConfigActionDefault = ""
+	gitConfigActionUpdate  = "update"
+	gitConfigActionReset   = "reset"
+
+	// gitCoreSSHCommand is the Git config used for setting up alternative SSH
+	// command. For Git-proxying, the command should point to "tsh git ssh".
+	//
+	// https://git-scm.com/docs/git-config#Documentation/git-config.txt-coresshCommand
+	gitCoreSSHCommand = "core.sshcommand"
+)
+
+func newGitConfigCommand(parent *kingpin.CmdClause) *gitConfigCommand {
+	cmd := &gitConfigCommand{
+		CmdClause: parent.Command("config", "Check Teleport config on the working Git directory. Or provide an action ('update' or 'reset') to configure the Git repo."),
+	}
+
+	cmd.Arg("action", "Optional action to perform. 'update' to configure the Git repo to proxy Git commands through Teleport. 'reset' to clear Teleport configuration from the Git repo.").
+		EnumVar(&cmd.action, gitConfigActionUpdate, gitConfigActionReset)
+	return cmd
+}
+
+func (c *gitConfigCommand) run(cf *CLIConf) error {
+	// Make sure we are in a Git dir.
+	err := execGitWithStdoutAndStderr(cf, io.Discard, io.Discard, "rev-parse", "--is-inside-work-tree")
+	if err != nil {
+		// This error message is a slight alternation of the original error
+		// message from the above command.
+		return trace.BadParameter("the current directory is not a Git repository (or any of the parent directories)")
+	}
```

Comment:
I think that without `git` installed this will give incorrect error message ("not a Git repo" instead of "git not found").